Remove all experiments (#2722)
iterative/dvc.org: content/docs/command-reference/exp/remove.md

## Synopsis

```usage

> usage: dvc experiments remove [-h] [-q | -v] [--queue | -A]
>                       [<experiment> [<experiment> ...]]

Reviewer: Why change these parts? 🙂 breaks aligning too
```suggestion
usage: dvc exp remove [-h] [-q | -v] [--queue | -A]
                      [experiment [experiment ...]]
```

Author: I copied the usage line from 
```
dvc exp remove --help
```
I guess we need to fix it there.

Reviewer: Yeah I imagined that was it, no worries. We don't need to have the exact same thing in both places. (May be hard to customize in Python argparse). Ok I'm committing this too...

Reviewer: > I guess we need to fix it there.

p.s. @karajan1001 well if it shows `<experiment>` as the argument name I think that should be changed in the core repo, yes. We only use `<name>` for option argument metavars I think.

Reviewer: This one matter left here though @karajan1001 ☝️

- `--queue` - remove all experiments that haven't been run yet (defined via
  `dvc exp run --queue`).

> - `-A`, `--all` - remove all experiments in this project.

Reviewer: Why do we specify "in this project"? Is there a way to `exp remove` from remotes?

Reviewer: Also, does it include queued ones? If so
```suggestion
- `-A`, `--all` - remove all experiments (includes `--queue`).
```

Author: @jorgeorpinel 
Not included for now.

Author: > Why do we specify "in this project"? Is there a way to exp remove from remotes? Otherwise

Yes there is one, https://github.com/iterative/dvc/pull/6471.

Reviewer: > Yes there is one, iterative/dvc#6471.

OK but that's not merged yet so unrelated to this PR 🙂 so let's not make the distinction yet but we'll revisit very soon per iterative/dvc#6471 😬 . Committing 2nd suggestion (resolving).

- `--queue` - remove all experiments that haven't been run yet (defined via
  `dvc exp run --queue`).

> - `-A`, `--all` - remove all experiments in this repo(defined via
>   `dvc exp run -A`).

Reviewer: ```suggestion
- `-A`, `--all` - remove all experiments in this project.
```

        exp-e6c97
        exp-1dad0
        exp-1df77
        exp-23d5a
```

> To remove any of them, just give their names to `dvc exp remove`:

Reviewer: ```suggestion
To remove any of them, give their names to `dvc exp remove`. Or use the `--all`
(`-A`) option to remove them all at once:
```

$ dvc exp list
master:
        exp-e6c97
        exp-23d5a
$ dvc exp remove -A

$ dvc exp list

$

> ```

Reviewer: This needs explaining and removing some of the last lines please @karajan1001

Author: How to fix this?

Reviewer: ````suggestion
$ dvc exp list
master:
        exp-e6c97
        exp-23d5a

$ dvc exp remove -A

$ dvc exp list
```

Nothing is listed after the last `dvc exp list` because they're all gone.
````
